packages/obonode/obojobo-chunks-iframe/render-settings.js

```diff
@@ -47,6 +47,9 @@ const getSetDimensions = (modelState, defaultWidth, defaultHeight) => ({
 })
 
 const getScaleAmount = (actualWidth, padding, setWidth) => {
+	if (actualWidth > setWidth) {
```

Comment:
This causes issues with fixed iframes however. I embedded a webpage at 640x480 Fixed and the iframe started at around a 3x zoom with this change.

Reply:
I think your answer might be on the line `const scaleAmount = getScaleAmount(actualWidth, padding, setDimensions.w)`. The last value passed in, `setWidth`, is supposed to represent the actual iframe width, which before this change was always what the user typed in. So if they put in 640 x 480 then `const scaleAmount = getScaleAmount(actualWidth, padding, setDimensions.w)` would be `const scaleAmount = getScaleAmount(actualWidth, padding, 640)`. But with these two new sizing options that is no longer true. Max size, for example, is about 834px. If I hardcode that value then the max size iframes seem to be scaling as I'd expect.

So I think maybe `const scaleAmount = getScaleAmount(actualWidth, padding, setDimensions.w)` needs to be changed such that the last parameter is given different values if the sizing of the iframe is set to MAX or TEXT. As much as I hate to hardcode magic values, I think that's probably the best answer here. If, for example, the sizing of the iframe is set to MAX then we can use a const like `WIDTH_MAX = 834` and pass it in instead of `setDimensions.w`.

Hopefully that makes sense, this code is pretty gnarly.

Reply:
That makes sense. I hadn't tested the changes with the `fixed` value. I was also thinking about implementing your idea with the hardcoded values before yesterday :) I will let you know how it goes.